pkg/ottl/ottlfuncs/func_contains.go

```diff
@@ -0,0 +1,46 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type ContainsArguments[K any] struct {
+	Target []ottl.StringLikeGetter[K]
```

Comment:
this won't allow you to pass a slice, consider this use case
`contains(attributes["tags"], "staging")`

Reply:
Maybe I'm wrong, but I tried to write e2e test and it seems that the following statement works and I can pass the slices:

```
set(attributes["test"], "pass") where Contains(["hello", "world"], "hello")
```

Did I miss anything?

Reply:
the thing is you're just assuming StringLikeGetter, 
when I want to use config like this
```yaml
 - context: log
        statements: 
          - set(attributes["tags"], ["my funky value 2", "staging", "hello", "world", "work"])
          - set(attributes["staging"], "true") where Contains(attributes["tags"], "staging")
```

Contains fail because first attribute is not an array.
so i need to fix it like this
```yml
    log_statements:
      - context: log
        statements: 
          - set(attributes["tags"], ["staging", "hello", "world", "work"])
          - set(attributes["staging"], "true") where Contains([attributes["tags"]], "staging")
```

by wrapping `attributes["tags"]` with `[]` to make it an array.
when i do that, your StringLikeGetter will get the `attributes["tags"]` and finds out it's a Slice. So it makes a string out of slice by [marshaling the array](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e96879b4058fe2f8887e1670e701d21cc5d78d26/pkg/ottl/expression.go#L409). 
Now you have marshaled array as a string`["staging", "hello", "world", "work"]` that does not equal `staging` in my condition. 

What you need to do is to use Getter and work with different types you may get. You may get `Slice`, `Map` or a string wrapped in `pcommon.Value`.  You should handle all of these differently

Reply:
I really appreciate the detailed answer that helps me understand the flow. Thanks.

I added e2e with the scenario you described. I hope it is better now.